### PR TITLE
tests: switch from fork to upgrade logic

### PR DESCRIPTION
**Pull request by czarcas7ic**
Merge after https://github.com/osmosis-labs/osmosis/pull/1809

## What is the purpose of the change

Our current e2e test framework upgrades from v8 to v11. The above PR introduced fork logic, however we (hopefully) aren't going to be forking in the next upgrade. This PR utilizes normal upgrade logic to upgrade from v10 to v11, while introducing a bug fix that was introduced from the above PR (relating to negative fork height when fork height was 0)

## Testing and Verifying

This change is already covered by existing tests

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? no
  - How is the feature or change documented? not applicable

### CODE REVIEW

File: tests/e2e/e2e_setup_test.go [hunk 1]
```diff
@@ -233,7 +231,8 @@ func (s *IntegrationTestSuite) SetupSuite() {
 		}
 	}
 
-	if !s.skipUpgrade {
+	switch {
```
Comment (p0mvn) [inline tests/e2e/e2e_setup_test.go:234]: Were you trying things here? I'm not sure what this change is about

Comment (czarcas7ic) [reply tests/e2e/e2e_setup_test.go:234]: i think its a cherry pick issue, my bad

File: tests/e2e/e2e_setup_test.go [hunk 2]
```diff
@@ -405,7 +403,10 @@ func (s *IntegrationTestSuite) configureChain(chainId string, validatorConfigs [
 		return
 	}
 
-	s.forkHeight = s.forkHeight - forkHeightPreUpgradeOffset
+	if s.isFork {
```
Comment (alexanderbez) [inline tests/e2e/e2e_setup_test.go:406]: We should add a small comment as to what we're doing here and why.

**Review by p0mvn — APPROVED** [review]
LGTM

File: tests/e2e/e2e_setup_test.go (hunk 2 above)
Comment (czarcas7ic) [reply tests/e2e/e2e_setup_test.go:406]: oops, I merged before addressing this. I will address in the next PR. Or @p0mvn if you do an e2e pr before me I can shoot you a comment to add here as well

Comment (p0mvn) [reply tests/e2e/e2e_setup_test.go:406]: Sounds good. I'll make an issue, whoever opens a PR next can add the comment and close the issue.

Comment (p0mvn) [reply tests/e2e/e2e_setup_test.go:406]: https://github.com/osmosis-labs/osmosis/issues/1869

**Pull request merged**